lib/srv/db/cloud/rds.go

```diff
@@ -0,0 +1,198 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+import (
+	"context"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/cloud"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/aws/aws-sdk-go/service/rds/rdsiface"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+// rdsConfig is the config for the client that auto-configures RDS databases.
+type rdsConfig struct {
+	// clients is an interface for creating AWS clients.
+	clients common.CloudClients
+	// identity is AWS identity this database agent is running as.
+	identity cloud.AWSIdentity
+	// database is the database instance to configure.
+	database types.Database
+}
+
+// Check validates the config.
+func (c *rdsConfig) Check() error {
+	if c.clients == nil {
+		return trace.BadParameter("missing parameter clients")
+	}
+	if c.identity == nil {
+		return trace.BadParameter("missing parameter identity")
+	}
+	if c.database == nil {
+		return trace.BadParameter("missing parameter database")
+	}
+	return nil
+}
+
+// newRDS creates a new RDS configurator.
+func newRDS(ctx context.Context, config rdsConfig) (*rdsClient, error) {
+	if err := config.Check(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	rds, err := config.clients.GetAWSRDSClient(config.database.GetAWS().Region)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	iam, err := config.clients.GetAWSIAMClient(config.database.GetAWS().Region)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &rdsClient{
+		cfg: config,
+		rds: rds,
+		iam: iam,
+		log: logrus.WithFields(logrus.Fields{
+			trace.Component: "rds",
+			"db":            config.database.GetName(),
+		}),
+	}, nil
+}
+
+type rdsClient struct {
+	cfg rdsConfig
+	rds rdsiface.RDSAPI
+	iam iamiface.IAMAPI
+	log logrus.FieldLogger
+}
+
+// setupIAM configures IAM for RDS or Aurora database.
+func (r *rdsClient) setupIAM(ctx context.Context) error {
+	var errors []error
+	if err := r.ensureIAMAuth(ctx); err != nil {
+		if trace.IsAccessDenied(err) { // Permission errors are expected.
+			r.log.Debugf("No permissions to enable IAM auth: %v.", err)
+		} else {
+			errors = append(errors, err)
+		}
+	}
+	if err := r.attachIAMPolicy(ctx); err != nil {
+		if trace.IsAccessDenied(err) { // Permission errors are expected.
+			r.log.Debugf("No permissions to attach IAM policy: %v.", err)
```

Comment:
So the reason it's handled like this is because access denied is an expected error (if somebody doesn't want to grant additional permissions to use auto-config) so I wanted to log it in a non-scary way to highlight that this is a normal scenario. Debug level was also picked for the same reason - otherwise if somebody isn't using auto-IAM, we'd spam their logs every time we try to auto-configure IAM and fail.